### PR TITLE
Feature global categories

### PR DESCRIPTION
**Pull request by Fasse**
#616

### CODE REVIEW

File: adm_program/system/classes/tabledate.php [hunk 1]
```diff
@@ -110,16 +110,15 @@ public function editable()
...
             {
                 return true;
             }
 
-            // Global events could be edited by the parent organization
-            if ($gCurrentOrganization->isChildOrganization($catOrgId) && $this->getValue('dat_global'))
+            // parent organizations could edit global events,
+            // child organizations could only edit their own events
+            if ($gCurrentOrganization->isParentOrganization()
+            || ($gCurrentOrganization->isChildOrganization() && (int) $gCurrentOrganization->getValue('org_id') == (int) $this->getValue('cat_org_id')))
```
Comment (thomas-rcv) [inline adm_program/system/classes/tabledate.php]: Add a tab

File: adm_program/system/classes/tableweblink.php [hunk 2]
```diff
@@ -33,15 +33,32 @@ public function __construct(&$database, $lnkId = 0)
...
+        {
+            if ($gCurrentOrganization->countAllRecords() === 1)
+            {
+                return true;
+            }
+
+            // parent organizations could edit global weblinks,
+            // child organizations could only edit their own weblinks
+            if ($gCurrentOrganization->isParentOrganization()
+            || ($gCurrentOrganization->isChildOrganization() && (int) $gCurrentOrganization->getValue('org_id') == (int) $this->getValue('cat_org_id')))
```
Comment (thomas-rcv) [inline adm_program/system/classes/tableweblink.php]: Add a tab for condition spacing

File: demo_data/db.sql [hunk 3]
```diff
@@ -55,8 +55,8 @@ create table %PREFIX%_announcements
 (
     ann_id                         integer       unsigned not null AUTO_INCREMENT,
     ann_cat_id                     integer       unsigned not null,
-    ann_global                     boolean       not null default '0',
     ann_headline                   varchar(100)  not null,
+    ann_global                  boolean             not null    default '0',
```
Comment (thomas-rcv) [inline demo_data/db.sql]: Spacing

File: demo_data/db.sql [hunk 4]
```diff
@@ -162,8 +162,8 @@ create table %PREFIX%_dates
     dat_id                         integer       unsigned not null AUTO_INCREMENT,
     dat_cat_id                     integer       unsigned not null,
     dat_rol_id                     integer       unsigned,
+    dat_global                     boolean             not null    default '0',
```
Comment (thomas-rcv) [inline demo_data/db.sql]: Spacing

File: adm_program/modules/categories/categories_new.php [hunk 5]
```diff
@@ -148,6 +148,12 @@
     {
         // a new category will be visible for all users per default
         $roleViewSet = array(0);
+
+        // profile fields should be organization independent all other categories should be organization dependent as defauld
```
Comment (thomas-rcv) [inline adm_program/modules/categories/categories_new.php]: Change dafauld -> default

**Comment by thomas-rcv** [comment]
Have a look to my comments. Generelly the code is ok.

**Comment by Fasse** [comment]
@thomas-rcv did you also do some basic tests?

**Comment by thomas-rcv** [comment]
Yes, of course. Couldn´t find Problems so far.
You can merge

**Pull request merged**